### PR TITLE
Update pipeline-service

### PR DESCRIPTION
**Pull request by Roming22**
* OSP-1.10
* PaC deployed through OSP-1.10
* allow for watcher logging level customization via gitops

### CODE REVIEW

**Comment by gabemontero** [comment]
I would think we would want to remove tekton-chains at https://github.com/openshift-pipelines/pipeline-service/tree/main/operator/gitops/argocd/pipeline-service/tekton-chains as well @Roming22 and validate that removal via pipeline service CI before updating staging 

Admittedly though, I'm a little unclear about how we override openshift-pipelines over at https://github.com/openshift-pipelines/pipeline-service/tree/main/operator/gitops/argocd/pipeline-service/openshift-pipelines so as to install or not install tekton-chains

of course, maybe I'm wrong here .... can you elaborate ?

thanks

**Comment by gabemontero** [comment]
> @gabemontero I'm not sure I understand why removing tekton-chains is a pre-requiste to upgrading to OSP-1.10.
> 
> We certainly want to remove it, sooner than later, but it's going to take some time before it's ready, and if possible I'd like to validate that OSP-1.10 is running smoothly as soon as possible to avoid last minute surprises before the freeze.

I thought PaC and tekton-chains were the same in that we had to install them from upstream to get required levels, but at 1.10 both the PaC and tekton-chains installed by openshift-pipelines were at sufficient levels

What motivated the removal of PaC via https://github.com/openshift-pipelines/pipeline-service/pull/620 to be included, bu not chains ... I would think we would want to be consistent.

Please clarify when you get the chance, but I won't block merge, assuming e2e's pass (they did not on the first go)

/lgtm

**Comment by gabemontero** [comment]
and the e2e failed with a gitops hiccup wrt pipeline service: 

```
 pipeline-service-in-cluster-local                 Unknown     Healthy
pipeline-service-in-cluster-local failed with:
[{"lastTransitionTime":"2023-04-25T20:26:09Z","message":"rpc error: code = Unknown desc = `kustomize build .components/pipeline-service/development` failed exit status 1: Error: accumulating resources: accumulation err='accumulating resources from 'git::https://github.com/openshift-pipelines/pipeline-service.git/developer/openshift/gitops/argocd/pipeline-service-storage?ref=14431236de0f1e3d3a9e27c145197d1a45942794': evalsymlink failure on '.components/pipeline-service/development/git::https:/github.com/openshift-pipelines/pipeline-service.git/developer/openshift/gitops/argocd/pipeline-service-storage?ref=14431236de0f1e3d3a9e27c145197d1a45942794' : lstat .components/pipeline-service/development/git::https:: no such file or directory': evalsymlink failure on '/tmp/kustomize-2974330201/developer/openshift/gitops/argocd/pipeline-service-storage' : lstat /tmp/kustomize-2974330201/developer/openshift/gitops/argocd/pipeline-service-storage: no such file or directory","type":"ComparisonError"}]
```

at first quick glance can't say for certain if it is related to the bump ...

/test appstudio-e2e-tests

**Comment by Mo3m3n** [comment]
@Roming22 can you update the commit message to add "using tls for rds connections" as result of this update ?

**Comment by gabemontero** [comment]
@Roming22 - unfortunately the latest run https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/redhat-appstudio_infra-deployments/1725/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1650981219980021760 failed in the same way as what I reported in https://github.com/redhat-appstudio/infra-deployments/pull/1725#issuecomment-1522401433

I think that points to gitops / kustomize is not happy with something in pipeline-service between `062d00d304d6fba316085784f3b948217fc99d12` and `14431236de0f1e3d3a9e27c145197d1a45942794`

By rough count I see 21 or 22 commits between those 2 commits, so if I am correct there is a large set to choose from, unless the error message wrt `pipeline-service-storage` helps narrow it down

**Comment by adambkaplan** [comment]
The external secrets for Pipelines as Code probably need to be updated. I think the PR as is will break staging because PaC will be deployed in the `openshift-pipelines` namespace.

File: components/pipeline-service/development/kustomization.yaml [hunk 1]
```diff
@@ -8,8 +8,8 @@ commonAnnotations:
   argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
 
 resources:
-  - git::https://github.com/openshift-pipelines/pipeline-service.git/developer/openshift/gitops/argocd/pipeline-service?ref=062d00d304d6fba316085784f3b948217fc99d12
-  - git::https://github.com/openshift-pipelines/pipeline-service.git/developer/openshift/gitops/argocd/pipeline-service-storage?ref=062d00d304d6fba316085784f3b948217fc99d12
+  - git::https://github.com/openshift-pipelines/pipeline-service.git/developer/openshift/gitops/argocd/pipeline-service?ref=14431236de0f1e3d3a9e27c145197d1a45942794
+  - git::https://github.com/openshift-pipelines/pipeline-service.git/developer/openshift/gitops/argocd/pipeline-service-storage?ref=14431236de0f1e3d3a9e27c145197d1a45942794
```
Comment (Mo3m3n) [inline components/pipeline-service/development/kustomization.yaml:12]: @Roming22 This is not going to work since pipeline-service-storage is not available anymore in pipeline-service repo it was replaced by two apps: minio + postgres.

I have sent this PR https://github.com/openshift-pipelines/pipeline-service/pull/621 to remediate to that.

**Review by gabemontero — APPROVED** [review]
/lgtm

**Comment by Roming22** [comment]
@Mo3m3n @adambkaplan Comments processed. SHA updated, and there's no references to the `pipelines-as-code` namespace anymore.

```
> kustomize build components/pipeline-service/staging/base | grep -c "namespace: pipelines-as-code"
0
```

**Comment by gabemontero** [comment]
pr needs a rebase @Roming22 

do you want to do it, or I can hit the "update with rebase" button if you like

File: components/pipeline-service/staging/base/kustomization.yaml [hunk 2]
```diff
@@ -20,4 +20,9 @@ patches:
     target:
       kind: ConfigMap
       name: pipelines-as-code
+      namespace: openshift-pipelines
+  - path: pac-app-secret.yaml
+    target:
+      kind: ExternalSecret
+      name: pipelines-as-code-secret
       namespace: pipelines-as-code
```
Comment (Mo3m3n) [inline components/pipeline-service/staging/base/kustomization.yaml:28]: @Roming22 we still have this one that should be replaced with openshift-pipelines probably

Comment (Mo3m3n) [reply components/pipeline-service/staging/base/kustomization.yaml:28]: Ah no that's irrelevant, it is actually the patch you made to update the ns.
I though it would be also better to fix the ns directly here https://github.com/redhat-appstudio/infra-deployments/blob/main/components/pipeline-service/base/external-secrets/pipelines-as-code/kustomization.yaml. However we can do that only when prod is also updated this way we can have the right ns directly in the base files

Comment (Roming22) [reply components/pipeline-service/staging/base/kustomization.yaml:28]: @Mo3m3n I don't believe so. I cannot change the namespace directly in `components/pipeline-service/base/external-secrets/pipelines-as-code/pipelines-as-code-secret.yaml` as it would impact production as well.

Therefore I need to apply a patch to the resource that is initially in the old namespace. It is in the right namespace after the patch is applied.
```
kind: ExternalSecret
metadata:
  annotations:
    argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
    argocd.argoproj.io/sync-wave: "-1"
  name: pipelines-as-code-secret
  namespace: openshift-pipelines
spec:
  dataFrom:
  - extract:
      key: staging/pipeline-service/github-app
```

Comment (Mo3m3n) [reply components/pipeline-service/staging/base/kustomization.yaml:28]: yes that's why I have concluded with:
> However we can do that only when prod is also updated this way we can have the right ns directly in the base files

**Comment by Mo3m3n** [comment]
@Roming22 I have just found that you also need to update this line https://github.com/redhat-appstudio/infra-deployments/blob/main/hack/secret-creator/create-plnsvc-secrets.sh#L27 to match this change done here in pipeline-service https://github.com/openshift-pipelines/pipeline-service/commit/b5f1dc20c3dd8980825c2255cacc85da77ea3460#diff-f394b1c9744cc285b6ac80384e967bd73210dd72a905e1ad7581b8ecab4e3773R201

Basically when switching to the helm based postgres deployment the name of the db service changed too (for dev mode). That's why I suggested in my [comment](https://github.com/redhat-appstudio/infra-deployments/pull/1725#issuecomment-1522415547) to also add to the commit message that we are enabling tls with this update and using Helm for postgres deployment

**Comment by Roming22** [comment]
Tests are failing because the `build` component is referencing the `pipelines-as-code` namespace. Waiting on a new version of `build` from @psturc that will be use the `openshift-pipelines` namespace instead.

The Pr will need to be updated to include the new build version.

**Comment by Michkov** [comment]
Another places where the code should be changed due to change of namespace:
- hack/build/setup-pac-integration.sh
- components/authentication/base/build.yaml ?
- components/cluster-secret-store/base/appsre-stonesoup-vault-secret-store.yaml ?
- components/sprayproxy/staging/add-backends.yml
(with ? should probably stay for now to not break prod -> but could cause some issues on stage)

**Comment by Mo3m3n** [comment]
It seems CI is still failing due to PAC
```  
 [mvp-demo-suite MVP Demo tests] MVP Demo Chapter 2 - advanced pipeline, JVM rebuild, successful release [It] upgrading to SLSA level 3 customizable pipeline triggers creation of a PR in the sample repo [mvp-demo, mvp-demo-chapter-2]
/tmp/tmp.AyzcdtidFU/tests/mvp-demo/mvp-demo.go:344 
 
[FAILED] Timed out after 300.000s.
  timed out when waiting for init PaC PR to be created 
```
And looking into infra-deployment code there are still references to the old PAC NS:
https://github.com/redhat-appstudio/infra-deployments/blob/5f86b503a2126576b79b0c407752a09234bc11f2/hack/build/setup-pac-integration.sh#L51
https://github.com/redhat-appstudio/infra-deployments/blob/5f86b503a2126576b79b0c407752a09234bc11f2/hack/build/setup-pac-integration.sh#L81

Also components like [authentication](https://github.com/redhat-appstudio/infra-deployments/tree/main/components/authentication), [sprayproxy](https://github.com/redhat-appstudio/infra-deployments/tree/main/components/sprayproxy) they still have `pipelines-as-code` ns hardcoded

**Comment by gabemontero** [comment]
> @Mo3m3n - we minimally need #1737 to merge in order to pick up @mmorhun 's fix for with namespace PaC is installed in. It has not merged as I type this comment.
> 

@Roming22 just pulled ^^ into this PR :-) 

> For what you noted with https://github.com/redhat-appstudio/infra-deployments/tree/main/components/authentication and https://github.com/redhat-appstudio/infra-deployments/tree/main/components/sprayproxy yes that is part of what @Michkov noted in [#1725 (comment)](https://github.com/redhat-appstudio/infra-deployments/pull/1725#issuecomment-1526126895)
> 
> I would say @Roming22 each of those need to be addressed as part of this PR
> 
> Once all that happens, we can see about rebase/retest and try to merge

**Comment by Roming22** [comment]
@Michkov I believe I've updated all references to pipelines-as-code. I tried to make sure not to impact production by patching where necessary.

**Comment by gabemontero** [comment]
the last CI failure needs https://github.com/redhat-appstudio/infra-deployments/pull/1579

we just (possibly re-)triaged that mvp-demo failure in https://redhat-internal.slack.com/archives/C02FANRBZQD/p1682697746956539

**Comment by gabemontero** [comment]
hopefully https://github.com/redhat-appstudio/e2e-tests/pull/488 fixes this

**Comment by gabemontero** [comment]
more new unrelated flakes being discussed in slack (I've asked QE to open blocking bugs and put skips in these failing tests until they are sorted out).

/retest

**Comment by gabemontero** [comment]
the latest failure is in some of the pipelines as code e2e code, so in theory could be related to this bump, as we are pulling in a new version of PaC, one which is installed by the openshift pipelines operator instead of upstream yaml


the pac logs, especially the controller, do not have a lot in them

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_infra-deployments/1725/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1652307069509832704/artifacts/appstudio-e2e-tests/redhat-appstudio-hypershift-gather/artifacts/pods/openshift-pipelines_pipelines-as-code-controller-5866597c4c-mjp5l_pac-controller.log
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_infra-deployments/1725/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1652307069509832704/artifacts/appstudio-e2e-tests/redhat-appstudio-hypershift-gather/artifacts/pods/openshift-pipelines_pipelines-as-code-watcher-7848847bf6-qxvr4_pac-watcher.log
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_infra-deployments/1725/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1652307069509832704/artifacts/appstudio-e2e-tests/redhat-appstudio-hypershift-gather/artifacts/pods/openshift-pipelines_pipelines-as-code-webhook-6b4f79686c-lgvcl_pac-webhook.log

*maybe* github throttling but is hard to say

**Comment by gabemontero** [comment]
I see the same `timed out when waiting for init PaC PR to be created` error @Roming22  .... with minimal activity in the PaC pods

By comparison, https://github.com/redhat-appstudio/infra-deployments/pull/1741 is showing green with the older version of PaC.

Might be time to try out this locally, work with @psturc and @flacatus to understand the details of this test, see where the breakdown is.

**Comment by Michkov** [comment]
Issue in build-service:
```
{"level":"error","ts":"2023-04-30T23:29:58.624Z","logger":"ComponentOnboarding.PaC-setup","caller":"controllers/component_build_controller_pac.go:415","msg":"failed to get Component PaC repository object","controller":"component","controllerGroup":"appstudio.redhat.com","controllerKind":"Component","Component":{"name":"test-mvp-component-kpta","namespace":"mvp-demo-dev-namespace-cbpg-tenant"},"namespace":"mvp-demo-dev-namespace-cbpg-tenant","name":"test-mvp-component-kpta","reconcileID":"9a13383c-095c-4cdc-91f8-a7a4898f6f74","action":"VIEW","error":"no matches for kind \"Repository\" in version \"pipelinesascode.tekton.dev/v1alpha1\"","stacktrace":"github.com/redhat-appstudio/build-service/controllers.(*ComponentBuildReconciler).ensurePaCRepository\n\t/opt/app-root/src/controllers/component_build_controller_pac.go:415\ngithub.com/redhat-appstudio/build-service/controllers.(*ComponentBuildReconciler).ProvisionPaCForComponent\n\t/opt/app-root/src/controllers/component_build_controller_pac.go:113\ngithub.com/redhat-appstudio/build-service/controllers.(*ComponentBuildReconciler).Reconcile\n\t/opt/app-root/src/controllers/component_build_controller.go:297\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:122\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:323\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:274\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/opt/app-root/src/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.1/pkg/internal/controller/controller.go:235"}
```

File: components/pipeline-service/development/kustomization.yaml (hunk 1 above)
Comment (Roming22) [reply components/pipeline-service/development/kustomization.yaml]: Version updated to include the fix.

**Comment by Michkov** [comment]
```{"severity":"error","timestamp":"2023-05-02T16:51:35.531Z","logger":"pipelines-as-code-webhook.ValidationWebhook","caller":"controller/controller.go:566","message":"Reconcile error","knative.dev/traceid":"78b06509-0c86-464b-8b97-034e745398df","knative.dev/key":"validation.pipelinesascode.tekton.dev","duration":0.000023802,"error":"secret \"pipelines-as-code-webhook-certs\" is missing \"ca-cert.pem\" key","stacktrace":"knative.dev/pkg/controller.(*Impl).handleErr\n\t/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/knative.dev/pkg/controller/controller.go:566\nknative.dev/pkg/controller.(*Impl).processNextWorkItem\n\t/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/knative.dev/pkg/controller/controller.go:543\nknative.dev/pkg/controller.(*Impl).RunContext.func3\n\t/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/knative.dev/pkg/controller/controller.go:491"}```
-> "error":"secret \"pipelines-as-code-webhook-certs\" is missing \"ca-cert.pem\" key"

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_infra-deployments/1725/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1653436401850519552/artifacts/appstudio-e2e-tests/redhat-appstudio-hypershift-gather/artifacts/pods/openshift-pipelines_pipelines-as-code-webhook-6b4f79686c-bsvr2_pac-webhook.log

**Comment by gabemontero** [comment]
So still the same error.

@Roming22 - so in my local repro of this, I killed the build-service pod, and when it restarted, it no longer got the `no matches for kind \"Repository\" in version \"[pipelinesascode.tekton.dev/v1alpha1](http://pipelinesascode.tekton.dev/v1alpha1)`errors and the Repository object was created in the test namespace (which I had left up) after it was missing during the test failure
 
I wonder if this is a startup order sort of thing, where the build-service initializes its client before the openshift-pipelines operator has "fully installed things", where as previously, PaC was getting created by pipeline-service directly

Here are the current sync-waves I see for the two:
```
gmontero ~/go/src/github.com/redhat-appstudio/infra-deployments  (main)$ find . -name "*.yaml" -exec grep -s -H sync-wave {} \;
./components/build-service/base/external-secrets/pipelines-as-code-secret.yaml:    argocd.argoproj.io/sync-wave: "-1"
./components/pipeline-service/base/external-secrets/pipelines-as-code/pipelines-as-code-secret.yaml:    argocd.argoproj.io/sync-wave: "-1"
./components/pipeline-service/base/external-secrets/tekton-results/tekton-results-database.yaml:    argocd.argoproj.io/sync-wave: "-1"
./components/pipeline-service/base/external-secrets/tekton-results/tekton-results-s3.yaml:    argocd.argoproj.io/sync-wave: "-1"
```

**Comment by gabemontero** [comment]
> `{"severity":"error","timestamp":"2023-05-02T16:51:35.531Z","logger":"pipelines-as-code-webhook.ValidationWebhook","caller":"controller/controller.go:566","message":"Reconcile error","knative.dev/traceid":"78b06509-0c86-464b-8b97-034e745398df","knative.dev/key":"validation.pipelinesascode.tekton.dev","duration":0.000023802,"error":"secret \"pipelines-as-code-webhook-certs\" is missing \"ca-cert.pem\" key","stacktrace":"knative.dev/pkg/controller.(*Impl).handleErr\n\t/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/knative.dev/pkg/controller/controller.go:566\nknative.dev/pkg/controller.(*Impl).processNextWorkItem\n\t/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/knative.dev/pkg/controller/controller.go:543\nknative.dev/pkg/controller.(*Impl).RunContext.func3\n\t/go/src/github.com/openshift-pipelines/pipelines-as-code/vendor/knative.dev/pkg/controller/controller.go:491"}` -> "error":"secret "pipelines-as-code-webhook-certs" is missing "ca-cert.pem" key"
> 
> https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/redhat-appstudio_infra-deployments/1725/pull-ci-redhat-appstudio-infra-deployments-main-appstudio-e2e-tests/1653436401850519552/artifacts/appstudio-e2e-tests/redhat-appstudio-hypershift-gather/artifacts/pods/openshift-pipelines_pipelines-as-code-webhook-6b4f79686c-bsvr2_pac-webhook.log

looks like we officially have multiple problems

**Comment by Roming22** [comment]
Closed in favor of #1769

We'll be taking baby steps:
* Deploy OSP 1.10 without any changes to PaC
* Update PaC and tekton-chains images to use images built by Red Hat
* Deploy tekton-results through the operator
* Deploy tekton-chains through the operator
